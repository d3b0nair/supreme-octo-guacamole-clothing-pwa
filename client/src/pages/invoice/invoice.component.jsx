import React, { useEffect } from "react";

import { connect } from "react-redux";

import { fetchOrderStart } from "../../redux/order/order.actions";
import OrderContainer from "../../components/order-item/order-item.container";

function InvoicePage(props) {
  const { fetchOrder } = props;
  useEffect(() => {
    fetchOrder(props.match.params.id);
  }, [fetchOrder, props.match.params.id]);
  return (
      <OrderContainer />
  );
}

const mapDispatchToProps = (dispatch) => ({
  fetchOrder: (urlParam) => dispatch(fetchOrderStart(urlParam)),
});

export default connect(null, mapDispatchToProps)(InvoicePage);
