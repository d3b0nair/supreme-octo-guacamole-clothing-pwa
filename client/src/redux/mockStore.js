import { createStore } from "redux";
import { persistReducer } from "redux-persist";
import { rootReducer, persistConfig } from "./root-reducer";
export const createMockStore = ({ mockState, mockReducers }) => {
  const combinedReducers = persistReducer(
    persistConfig,
    rootReducer(mockReducers)
  );
  return createStore(combinedReducers, mockState);
};
