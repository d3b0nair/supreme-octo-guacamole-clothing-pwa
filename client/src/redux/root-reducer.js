import { combineReducers } from "redux";
import { persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";

import userReducer from "./user/user.reducer";
import cartReducer from "./cart/cart.reducer";
import directoryReducer from "./directory/directory.reducer";
import shopReducer from "./shop/shop.reducer";
import checkoutReducer from "./checkout/checkout.reducer";
import stripeReducer from "./stripe/stripe.reducer";
import orderReducer from "./order/order.reducer"

const persistConfig = {
  key: "root",
  storage,
  whitelist: ["cart", "stripe"],
};
const rootReducer = combineReducers({
  user: userReducer,
  cart: cartReducer,
  checkout: checkoutReducer,
  directory: directoryReducer,
  shop: shopReducer,
  stripe: stripeReducer,
  selectedOrder: orderReducer
});
export default persistReducer(persistConfig, rootReducer);
