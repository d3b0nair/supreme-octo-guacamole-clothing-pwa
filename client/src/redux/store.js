import { createStore, applyMiddleware, compose } from "redux";
import { persistStore } from "redux-persist";
import logger from "redux-logger";
import createSagaMiddleware from "redux-saga";

import rootReducer from "./root-reducer";
import rootSaga from "./root-saga";

const sagaMiddleware = createSagaMiddleware();

const middlewares = [sagaMiddleware];
const composeEnhancers =
  (process.env.NODE_ENV !== "production" &&
    typeof window !== "undefined" &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
  compose;
if (process.env.NODE_ENV === "development") {
  middlewares.push(logger);
}
export const store = createStore(
  rootReducer,
  composeEnhancers(applyMiddleware(...middlewares))
);

sagaMiddleware.run(rootSaga);

export const persistor = persistStore(store);
