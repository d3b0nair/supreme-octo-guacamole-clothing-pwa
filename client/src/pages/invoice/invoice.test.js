import React from "react";
import {
  render,
  cleanup,
} from "@testing-library/react";
import { createMockStore } from "../../redux/mockStore";
import { InvoicePage } from "./invoice.component";
import { Provider } from "react-redux";

const mockFetchOrder = jest.fn();
const mockMatch = {
  params: { id: "invoice_id" },
};
const mockState = {
  selectedOrder: {
    isLoading: true,
  },
};
const mockReducer = (
  state = {
    isLoading: true,
  },
  action
) => state;
const store = createMockStore({
  state: mockState,
  reducers: { selectedOrder: mockReducer },
});
const mockProps = (otherProps) => ({
  ...otherProps,
  match: mockMatch,
  fetchOrder: mockFetchOrder,
});
const component = (mockProps) => (
  <Provider store={store}>
    <InvoicePage {...mockProps} />
  </Provider>
);

afterEach(cleanup);

describe("Invoice Page component", () => {
  it("should mount", () => {
    const { container } = render(component(mockProps()));
    expect(container).toMatchSnapshot();
  });
  it("should fetch selected order", () => {
    render(component(mockProps()));
    expect(mockFetchOrder).toHaveBeenCalled();
  });
});
