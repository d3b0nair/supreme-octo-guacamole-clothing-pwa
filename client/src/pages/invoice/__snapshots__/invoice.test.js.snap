// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice Page component should mount 1`] = `
<div>
  <div
    class="sc-bdvvtL kHeCUE"
  >
    <div
      class="sc-gsDKAQ bMJKwb"
    />
  </div>
</div>
`;
